gitlab/resource_gitlab_project_access_token_test.go

```diff
@@ -0,0 +1,300 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectAccessTokenExists("gitlab_project_access_token.bar", &pat),
+					testAccCheckGitlabProjectAccessTokenAttributes(&pat, &testAccGitlabProjectAccessTokenExpectedAttributes{
+						name:      "my project token",
+						scopes:    map[string]bool{"read_repository": true, "api": true, "write_repository": true, "read_api": true},
+						expiresAt: "2022-04-01",
+					}),
+				),
+			},
+			// Update the Project Access Token to change the parameters
+			{
+				Config: testAccGitlabProjectAccessTokenUpdateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectAccessTokenExists("gitlab_project_access_token.bar", &pat),
+					testAccCheckGitlabProjectAccessTokenAttributes(&pat, &testAccGitlabProjectAccessTokenExpectedAttributes{
+						name:      "my new project token",
+						scopes:    map[string]bool{"read_repository": false, "api": true, "write_repository": false, "read_api": false},
+						expiresAt: "2022-05-01",
+					}),
+				),
+			},
+			// Add a CICD variable with Project Access Token value
+			{
+				Config: testAccGitlabProjectAccessTokenUpdateConfigWithCICDvar(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectAccessTokenExists("gitlab_project_access_token.bar", &pat),
+					testAccCheckGitlabProjectVariableExists(ctx.client, "gitlab_project_variable.var"),
+					testAccCheckGitlabProjectAccessTokenAttributes(&pat, &testAccGitlabProjectAccessTokenExpectedAttributes{
+						name:      "my new project token",
+						scopes:    map[string]bool{"read_repository": false, "api": true, "write_repository": false, "read_api": false},
+						expiresAt: "2022-05-01",
+					}),
+				),
+			},
+			//Restore Project Access Token initial parameters
+			{
+				Config: testAccGitlabProjectAccessTokenConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectAccessTokenExists("gitlab_project_access_token.bar", &pat),
+					testAccCheckGitlabProjectAccessTokenAttributes(&pat, &testAccGitlabProjectAccessTokenExpectedAttributes{
+						name:      "my project token",
+						scopes:    map[string]bool{"read_repository": true, "api": true, "write_repository": true, "read_api": true},
+						expiresAt: "2022-04-01",
+					}),
+				),
+			},
+			//Destroy Project Access Token
+			{
+				Config: testAccGitlabProjectAccessTokenDestroyToken(rInt),
+				Check:  testAccCheckGitlabProjectAccessTokenDoesNotExist(&pat),
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabProjectAccessTokenDoesNotExist(pat *testAccGitlabProjectAccessTokenWrapper) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		//Unfortunately we need to wait a bit, since the API call doesn't wait to destroy the PAT and returns immediately
+		time.Sleep(3 * time.Second) // lintignore: R018 // TODO: Resolve this tfproviderlint issue
```

Comment:
This is not great. Can you use `gomega.Eventually` here?

Something like

```go
import (
	. "github.com/onsi/gomega"
)

...

return func(s *terraform.State) error {
	return InterceptGomegaFailure(func() {
		Eventually(func() error {
			// Your test code here.
		}).Should(Succeed())
...
```

See the `testAccCheckGitlabProjectApprovalRuleAttributes` for example: https://github.com/gitlabhq/terraform-provider-gitlab/blob/79a17cbecec558f274a0d9a32dc22bdeff5d0750/gitlab/resource_gitlab_project_approval_rule_test.go#L92